tests/test_bqpipeline.py

```diff
@@ -63,24 +77,116 @@ def test_create_job_config_destination(self):
+        self.assertEqual(result, bigquery.ScalarQueryParameter('test',
+                                                               'BYTES', b))
+
+        with self.assertRaises(KeyError):
+            bqpipeline.set_scalar_parameter('test', 1, typ='unsupported')
+
+    def test_array_parameters(self):
+        result = bqpipeline.set_list_parameter('test', ['abc'], typ='list')
+        self.assertEqual(result, bigquery.ArrayQueryParameter('test', 'STRING',
+                                                              ['abc']))
+
+        result = bqpipeline.set_list_parameter('test', [1, 2], typ='list')
+        self.assertEqual(result, bigquery.ArrayQueryParameter('test', 'INT64',
+                                                              [1, 2]))
+
+        with self.assertRaises(IndexError):
+            bqpipeline.set_list_parameter('test', [], typ='list')
+
+    def test_struct_parameters(self):
+        result = bqpipeline.set_dict_parameter('test', {'a': 'abc'},
+                                               typ='list')
+        self.assertEqual(result, bigquery.StructQueryParameter(
+            'test',
+            bigquery.ScalarQueryParameter('a', 'STRING', 'abc')))
+
+    def test_positional_parameters(self):
+        result = bqpipeline.set_scalar_parameter(None, 'abc', typ='str')
+        self.assertEqual(result, bigquery.ScalarQueryParameter(None,
+                                                               'STRING', 'abc'))
+
+        result = bqpipeline.set_scalar_parameter(None, 1, typ='int')
+        self.assertEqual(result, bigquery.ScalarQueryParameter(None,
+                                                               'INT64', 1))
+
+        result = bqpipeline.set_list_parameter(None, ['abc'], typ='list')
+        self.assertEqual(result, bigquery.ArrayQueryParameter(None, 'STRING',
+                                                              ['abc']))
+
+    def test_query_params(self):
+        bq = bqpipeline.BQPipeline(
+            job_name='testjob', default_project='testproject',
+            default_dataset='testdataset')
+        cfg = bq.create_job_config(batch=False, create=False, overwrite=False)
+        self.assertIsNone(bq.set_query_params([]))
+
+        # Test positional query parameters.
+        self.assertEqual(bq.set_query_params([1,]),
+                         [bigquery.ScalarQueryParameter(None, 'INT64', 1)])
+
+        # Test named query parameters.
+        self.assertEqual(bq.set_query_params({'test': 1}),
+                         [bigquery.ScalarQueryParameter('test', 'INT64', 1)])
+
+        # Test invalid query parameters.
+        self.assertIsNone(bq.set_query_params([1, [1, 'two']]))
+
+    def test_invalid_query_params(self):
+        bq = bqpipeline.BQPipeline(
+            job_name='testjob', default_project='testproject',
+            default_dataset='testdataset')
+        cfg = bq.create_job_config(batch=False, create=False, overwrite=False)
```

Comment:
why are we even making this job config? appears to be unused.

Reply:
Done.